pyproject.toml

```
@@ -1,6 +1,269 @@
+[build-system]
+build-backend = "hatchling.build"
+requires = [
+    "hatchling>=1.2.0"
+]
+
+[project]
+name = "elyra"
+description="Elyra provides AI Centric extensions to JupyterLab"
+maintainers = [
+    {name = "Elyra Maintainers"},
+]
+readme = "README.md"
+requires-python = ">=3.7"
+dependencies = [
+    "autopep8>=1.5.0",
+    "click>=8",  # elyra-ai/elyra#2579
+    "colorama",
+    "deprecation",
+    "entrypoints>=0.3",
+    "jinja2>=3",
+    "jsonschema>=3.2.0,<4.0",  # Cap from kfp
+    "jupyter_core>=4.6.0",
+    "jupyter_client>=6.1.7",
+    "jupyter-packaging>=0.10",
+    "jupyter_server>=1.7.0",
+    "jupyterlab>=3.4.6",  # comment out to use local jupyterlab
+    "jupyterlab-lsp>=3.8.0",  # comment out to use local jupyterlab
+    "jupyterlab-git~=0.32",  # Avoid breaking 1.x changes
+    "jupyter-resource-usage>=0.5.1",
+    "MarkupSafe>=2.1",
+    "minio>=7.0.0",
+    "nbclient>=0.5.1",
+    "nbconvert>=6.4.5",
+    "nbdime~=3.1",  # Cap from jupyterlab-git
+    "nbformat>=5.1.2",
+    "networkx>=2.5.1",
+    "papermill>=2.3.4",
+    "python-lsp-server[all]>=1.1.0",
+    "pyyaml>=5.3.1,<6.0",  # Cap from kfp
+    "requests>=2.25.1",
+    "rfc3986-validator>=0.1.1",
+    "tornado>=6.1.0",
+    "traitlets>=4.3.2",
+    "typing-extensions>=3.10,<5",  # Cap from kfp
+    "urllib3>=1.26.5",
+    "watchdog>=2.1.3",
+    "websocket-client",
+    "yaspin",
+    "kfp>=1.7.0,<2.0,!=1.7.2",  # We cap the SDK to <2.0 due to possible breaking changes
+    "pygithub",
+    "black>=22.8.0",
+]
+classifiers = [
+    "License :: OSI Approved :: Apache Software License",
+    "Operating System :: OS Independent",
+    "Topic :: Scientific/Engineering",
+    "Topic :: Scientific/Engineering :: Artificial Intelligence",
+    "Topic :: Software Development",
+    "Topic :: Software Development :: Libraries",
+    "Topic :: Software Development :: Libraries :: Python Modules",
+    "Programming Language :: Python :: 3.7",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+]
+dynamic = ['version']
+
+[project.urls]
+Documentation = "https://elyra.readthedocs.io/en/latest"
+Homepage = "https://github.com/elyra-ai/elyra"
+
+[project.optional-dependencies]
+test = [
+    "elyra-examples-kfp-catalog",
+    "git-python",
+    "importlib-resources",
+    "pytest>=5.4.1",
+    "pytest-console-scripts",
```

Could we go ahead and add `pytest_jupyter` to get ahead of the server 2.0 release?  Once 2.0 is released (slated for Monday), our pytests will fail because the jupyter server pytest plugins we use will be provided by `pytest_jupyter` rather than `jupyter_server` directly.  I could go ahead and deliver a PR, but it will be essentially moot since it will be updating a location that will be obsoleted by this PR.
```suggestion
    "pytest-console-scripts",
    "pytest_jupyter",
```
I've confirmed our tests with `pytest_jupyter` installed, along with the _existing_ (pre 2.0) `jupyter_server` installed, still work.  So the presence of `pytest_jupyter` does not interfere with older `jupyter_server` installations.

---

added